pkg/telemetry/events.go

```diff
@@ -269,12 +271,38 @@ func (t *telemetryService) TrackSubscribed(
 	})
 }
 
-func (t *telemetryService) TrackUnsubscribed(ctx context.Context, participantID livekit.ParticipantID, track *livekit.TrackInfo) {
+func (t *telemetryService) TrackSubscribeFailed(
+	ctx context.Context,
+	participantID livekit.ParticipantID,
+	trackID livekit.TrackID,
+	err error,
+	isUserError bool,
+) {
 	t.enqueue(func() {
-		prometheus.SubSubscribedTrack(track.Type.String())
+		prometheus.RecordTrackSubscribeFailure(err, isUserError)
 
 		room := t.getRoomDetails(participantID)
-		t.SendEvent(ctx, newTrackEvent(livekit.AnalyticsEventType_TRACK_UNSUBSCRIBED, room, participantID, track))
+		ev := newTrackEvent(livekit.AnalyticsEventType_TRACK_SUBSCRIBE_FAILED, room, participantID, &livekit.TrackInfo{
+			Sid: string(trackID),
+		})
+		ev.Error = err.Error()
+		t.SendEvent(ctx, ev)
+	})
+}
+
+func (t *telemetryService) TrackUnsubscribed(
+	ctx context.Context,
+	participantID livekit.ParticipantID,
+	track *livekit.TrackInfo,
+	shouldSendEvent bool,
+) {
+	t.enqueue(func() {
+		prometheus.RecordTrackUnsubscribed(track.Type.String())
+
+		if shouldSendEvent {
```

Comment:
Is this not going to be sent in migration cases? Do we want one sent for migration also to get the full picture?

Reply:
I think ideally we would not show unsubscribe or subscribe events in the case of migrations. I think as it stands it'll still log Subscriptions again.. and that seems ok.

Reply:
From a reconciler perspective, I am guessing we want to have the switch from `local` -> `remote` or vice-versa as an attempt + success/failure so that it can take corrective action if there is a problem during the switch.

Whether we translate that to telemetry is something to think about. Yeah, not showing migration in telemetry sounds reasonable. But, wondering if we should have additional events to signify switch of source track from `local` -> `remote` and vice-versa.

Reply:
yeah this would be something to think about.. though this PR is getting a bit large. let's think about adding this insight in a future one.